src/bot.ts

```diff
@@ -146,10 +138,10 @@ export class DiscordBot {
         if (webhookID) {
             // webhookID and user IDs are the same, they are unique, so no need to prefix _webhook_
             const name = member instanceof Discord.User ? member.username : member.user.username;
-            const nameId = new MatrixUser(`@${name}`).localpart;
-            return this.bridge.getIntentFromLocalpart(`_discord_${webhookID}_${nameId}`);
+            // TODO: We need to sanitize name
+            return this.bridge.getIntentForSuffix(`${webhookID}_${name}`);
```

Comment:
soru once wrote a santizing routine for the previous thing, unknowingly that it already did it automatically, we could easily add it in again
It is also currently used in mx-puppet-bridge, meaning it is not lost somewhere *yay*

Reply:
Hmm, previously we used QP encoding. I'm unsure if the one you linked https://github.com/Sorunome/mx-puppet-bridge/blob/master/src/util.ts#L37 will look very pretty.